minetest/minetest
# Refactor parts of Irrlicht's material system

This is the Minetest side of minetest/irrlicht#204 ("Refactor parts of the material system and set min filter to bilinear for 2D graphics"). The actual change is in the Irrlicht PR, this is "just" refactoring I did for that change.

Together with minetest/irrlicht#204, this PR fixes #13621. Because it exposes all OpenGL texture filtering options to Minetest, it also makes #13108 trivial to fix.

See minetest/irrlicht#204 for more information.

## To do

This PR is a Ready for Review.

## How to test

Use Minetest on all platforms and with all Irrlicht video drivers and verify that everything looks the same as before, with one exception: Images rendered with a lower resolution than their original resolution in GUIs look better. The difference is very visible with #13606, for example.
